src/pages/courts/[courtID]/index.tsx

```diff
@@ -0,0 +1,86 @@
+import {
+  Text,
+  Button,
+  Flex,
+  Heading,
+  Table,
+  TableContainer,
+  Tbody,
+  Td,
+  Tr,
+} from "@chakra-ui/react";
+import { useRouter } from "next/router";
+import { ICourt } from "@/interfaces/courtData";
+import { useEffect, useState } from "react";
+import { api } from "@/utils/axios";
+import { BiPencil } from "react-icons/bi";
+import formatDate from "@/utils/formatDate";
+import { idRouteHandler, routeHandler } from "@/utils/routeHandler";
+import { headerCellGenerator } from "@/utils/headerCellGenerator";
+import { ChevronLeftIcon } from "@chakra-ui/icons";
+
+const CourtDetail = () => {
+  const [courtData, setCourtData] = useState<ICourt>();
+  const [loading, setLoading] = useState(true);
+  const router = useRouter();
+  const courtId = router.query.courtID;
+  const apiUrl = `courts/${courtId}`;
+
+  useEffect(() => {
+    api(apiUrl as string, { method: "get" }).then(({ data }) => {
+      ["_id", "isDeleted"].forEach((e) => delete data[e]);
+      delete data.isDeleted;
+      data["createdAt"] = formatDate(data.createdAt);
+      data["updatedAt"] = formatDate(data.createdAt);
+      setCourtData(data);
+      setLoading(false);
+    });
+  }, []);
```

Comment:
It won't be so much different for function execution, but maybe using "async...await"  would be more consistent with other async function.